pennylane/beta/plugins/default_tensor.py

```diff
@@ -428,38 +422,167 @@ def ev(self, obs_nodes, wires, contraction_method="auto"):
+            wires (Sequence[Sequence[int]]): measured subsystems for each observable
 
-        The contracted tensor is stored in the ``_nodes`` dictionary under the key ``"contracted_state"``.
+         Returns:
+            complex: expectation value :math:`\expect{A} = \bra{\psi}A\ket{\psi}`
+        """
+        self._contract_premeasurement_network()
+        ket = self._nodes["contracted_state"]
+        bra = tn.conj(ket, name="Bra")
+
+        all_wires = tuple(range(self.num_wires))
+        meas_wires = []
+        # For wires which are measured, add edges between
+        # the ket node, the observable nodes, and the bra node
+        for obs_node, obs_wires in zip(obs_nodes, wires):
+            meas_wires.extend(obs_wires)
+            for idx, w in enumerate(obs_wires):
+                # Use convention that the indices of a tensor are ordered like
+                # [output_idx1, output_idx2, ..., input_idx1, input_idx2, ...]
+                output_idx = idx
+                input_idx = len(obs_wires) + idx
+                tn.connect(obs_node[input_idx], ket[w])  # A|psi>
+                tn.connect(bra[w], obs_node[output_idx])  # <psi|A
+        # unmeasured wires are contracted directly between bra and ket
+        for w in set(all_wires) - set(meas_wires):
+            tn.connect(bra[w], ket[w])
+
+        # At this stage, all nodes are connected, and the contraction yields a
+        # scalar value.
+        ket_and_observable_node = ket
+        for obs_node in obs_nodes:
+            ket_and_observable_node = tn.contract_between(obs_node, ket_and_observable_node)
+        return tn.contract_between(bra, ket_and_observable_node).tensor
+
+    def _ev_mps(self, obs_nodes, wires):
+        r"""Expectation value of observables on specified wires using a MPS representation.
 
-        Args:
-            contraction_method (str): The contraction method to be employed.
-                Possible choices are "auto", "greedy", "branch", or "optimal".
-                See TensorNetwork library documentation for more details.
+         Args:
+            obs_nodes (Sequence[tn.Node]): the observables as TensorNetwork Nodes
+            wires (Sequence[Sequence[int]]): measured subsystems for each observable
+         Returns:
+            complex: expectation value :math:`\expect{A} = \bra{\psi}A\ket{\psi}`
+        """
+        if any(len(wires_seq) > 2 for wires_seq in wires):
+            raise NotImplementedError(
+                "Multi-wire measurement only supported for nearest-neighbour wire pairs."
+            )
+        else:
+            if len(obs_nodes) == 1 and len(wires[0]) == 1:
+                # TODO: can measure multiple local expectation values at once,
+                # but this would require change of `expval` behaviour and
+                # refactor of `execute` logic from parent class
+                expval = self.mps.measure_local_operator(obs_nodes, wires[0])[0]
```

Comment:
Nice, is this a TensorNetwork method? Looks like its listed in the documentation, but has no docstring

Reply:
Yes. They also have a function to measure two-body correlators, but it had a [bug](https://github.com/google/TensorNetwork/issues/527). Should be fixed in next release

Reply:
😆